site/docs/v1/tech/migration-guide/azureadb2c.adoc

```diff
@@ -0,0 +1,894 @@
+    -o, --only-link-social-accounts  Link social accounts with no import.
+    -u, --users-file USERS_FILE      The exported JSON user data file from Azure AD B2C. Defaults to users.json.
+    -f FUSIONAUTH_URL,               The location of the FusionAuth instance. Defaults to http://localhost:9011.
+        --fusionauth-url
+    -k, --fusionauth-api-key API_KEY The FusionAuth API key.
+    -t TENANT_ID,                    The FusionAuth tenant id. Required if more than one tenant exists.
+        --fusionauth-tenant-id
+    -h, --help                       Prints this help.
+----
+
+For this script to work correctly, set the following switches, unless the defaults work for you:
+
+* `-u` should point to the location of the user export file you obtained from running the API export, unless the default works.
+* `-f` must point to your FusionAuth instance. If you are testing locally, it will probably be `\http://localhost:9011`.
+* `-k` needs to be set to the value of the API key created above.
+* `-t` should be set to the Id of the testing tenant created above.
+
+The `-o` and `-l` switches will attempt to create links for any social users (where the user authenticated via Google or another social provider) found in the user's data file.
+
+If you are loading social users, you must create the social providers in FusionAuth beforehand, or the links will fail. Additionally, creating a link is not currently optimized in the same way that loading users is. So it may make sense to import all the users in one pass (omitting the `-l` switch). Then, after the users are imported, create the links using the `-o` switch in a second pass.
+
+[NOTE.note]
+====
+The social account linking functionality will only work with FusionAuth versions above or equal to 1.28. The `fusionauth_client` library must be >= 1.28 too.
+====
+
+Running the script should produce an output like this:
+
+[source,shell,title=Import script output]
+----
+$ ruby ./import.rb -f http://localhost:9011 -k '...'
+FusionAuth Importer : Azure AD B2C
+ > User file: user-data.json
+ > Call FusionAuth to import users
+ > Import success
+Duplicate users 0
+Import complete. 2 users imported.
+----
+
+==== Enhancing the Script
+
+You may also want to migrate additional data. Currently, the following attributes are migrated:
+
+* `user_id`
+* `email`
+* `email_verified`
+* `username`
+* `insertInstant`
+* `registrations`, if supplied
+
+The migrated user will have the Azure AD B2C original user identities stored on the `user.data` object. If you have additional user attributes to migrate, review and modify the `map_user` method.
+
+You may also want to assign Roles, or associate users with Groups, by creating the appropriate JSON data structures in the import call. These are documented in the link:/docs/v1/tech/apis/users#import-users[Import User API docs]. This will require modifying the `import.rb` code.
+
+=== Verify the Import
+
+include::docs/v1/tech/migration-guide/_verify-import.adoc[]
```

Comment:
This includ file needs to be modified because "If you have a test user whose password you know, open an incognito window and log in to ensure the hash migration was successful." is not going to work, ever, because we are importing the password hashes as a random string. 

We should get rid of everything after "You can manage the user by clicking on the black button to the right of the Created date in the list to review the details of the imported user’s profile." if this is being used as part of a migration where you have to reset passwords. Or you can remove it, the Cognito guide doesn't have a 'verify import' section for the bulk import: https://fusionauth.io/docs/v1/tech/migration-guide/cognito#import-the-users